### PR TITLE
Jigsaw support

### PR DESCRIPTION
**Pull request by lpandzic**
https://github.com/FasterXML/jackson-databind/issues/1536

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Would it be possible to add bit of description of what changes are, and for why?

**Comment by lpandzic** [comment]
Sure.

File: pom.xml [hunk 1]
```diff
@@ -26,14 +26,14 @@
...
-    <javac.target.version>1.6</javac.target.version>
+    <maven.compiler.release>9</maven.compiler.release>
     <!-- 02-Jun-2016, tatu: Looks like newer versions (3.0+) of bundle plugin require Java 7 (directly
             or via transitive deps), so need to downgrade
       -->
-    <version.plugin.bundle>2.5.3</version.plugin.bundle>
 
     <osgi.export>com.fasterxml.jackson.annotation.*;version=${project.version}</osgi.export>
+    <maven-compiler-plugin.version>3.6.1</maven-compiler-plugin.version>
+    <maven-bundle-plugin.version>3.2.0</maven-bundle-plugin.version>
```
Comment (lpandzic) [inline pom.xml]: version upgrades are required for both in order to handle maven-module.java

File: pom.xml [hunk 2]
```diff
@@ -44,4 +44,27 @@
...
+    <plugin>
+      <groupId>org.apache.felix</groupId>
+      <artifactId>maven-bundle-plugin</artifactId>
+      <extensions>true</extensions>
+      <version>${maven-bundle-plugin.version}</version>
+      <configuration>
+        <instructions>
+          <_noee>true</_noee>
+          <_failok>true</_failok>
+          <Include-Resource>module-info.class=target/classes/module-info.class</Include-Resource>
```
Comment (lpandzic) [inline pom.xml]: required for bundle plugin to include module-info.class in built jar

File: pom.xml [hunk 3]
```diff
@@ -44,4 +44,27 @@
...
+      </plugin>
+    <plugin>
+      <groupId>org.apache.felix</groupId>
+      <artifactId>maven-bundle-plugin</artifactId>
+      <extensions>true</extensions>
+      <version>${maven-bundle-plugin.version}</version>
+      <configuration>
+        <instructions>
+          <_noee>true</_noee>
+          <_failok>true</_failok>
```
Comment (lpandzic) [inline pom.xml]: without failok build fails:
[ERROR] Bundle com.fasterxml.jackson.core:jackson-annotations:bundle:2.9.0-SNAPSHOT : Invalid class file module-info.class (java.lang.ArrayIndexOutOfBoundsException: 19)

File: pom.xml [hunk 4]
```diff
@@ -44,4 +44,27 @@
...
+        <version>${maven-compiler-plugin.version}</version>
+      </plugin>
+    <plugin>
+      <groupId>org.apache.felix</groupId>
+      <artifactId>maven-bundle-plugin</artifactId>
+      <extensions>true</extensions>
+      <version>${maven-bundle-plugin.version}</version>
+      <configuration>
+        <instructions>
+          <_noee>true</_noee>
```
Comment (lpandzic) [inline pom.xml]: without noee build fails:
[ERROR] An internal error occurred
java.lang.IllegalArgumentException: Null query

**Comment by cowtowncoder** [comment]
Ok so... does this mean that only JDK9 build would be enabled for Travis? But also that that version is not yet available?

**Comment by cowtowncoder** [comment]
Before JDK 9, what we can do is #123 which I can add for `2.9.1` (unusual change for patch version, but makes sense here).

**Comment by cowtowncoder** [comment]
Closing in favor of new attempt for 2.10

**Pull request closed without merging**